### ISSUE TITLE
Build failure with OCaml 4.07 + Coq master

### ISSUE THREAD

**Issue by ejgallego** [AbsInt/CompCert#241]
See log https://gitlab.com/coq/coq/-/jobs/80217808:
```ocaml
OCAMLC   lib/Camlcoq.ml
File "lib/Camlcoq.ml", line 23, characters 0-11:
Error (warning 44): this open statement shadows the module identifier Float (which is later used)
Makefile.extr:119: recipe for target 'lib/Camlcoq.cmo' failed
```

This could be also considered a Coq bug, submitting first here as to understand what do Compcert devs think.

**Comment by xavierleroy** [comment]
Well, it looks like one following Coq stdlib modules now defines a Float module?   Do you know which one?  And for which purpose?

 Datatypes
 BinNums
 BinNat
 BinInt
 BinPos

An `open! Floats` will fix the issue at little cost, but I'd like to understand what's happening.

**Comment by ejgallego** [comment]
@xavierleroy is OCaml 4.07 who has added the `Float` module file in ocaml/ocaml@7f6d059f0863f76d8ccd1f4a395646ab437440c6 , which now conflicts with the `Float` module of CompCert.

**Comment by xavierleroy** [comment]
Silly me!  Of course it's not a Coq problem but an OCaml problem.  Thanks for setting me on the right path.  Will come back with a fix very soon.

**Comment by xavierleroy** [comment]
Fixed by using `open! Floats`.

**Comment by ejgallego** [comment]
Thank you!

**Comment by ejgallego** [comment]
@xavierleroy note that some conflicts do still remain, example:
```
OCAMLOPT cfrontend/C2C.ml
File "cfrontend/C2C.ml", line 19, characters 0-11:
Error (warning 44): this open statement shadows the module identifier Float (which is later used)
```
unfortunately I cannot reopen this issue due to permissions.

I'd suggest to grep for all such occurences.

**Comment by xavierleroy** [comment]
Dear Emilio, thanks for the notice.  Indeed I haven't tested with OCaml 4.07 yet, but now that it is out I will!  Reopening...

**Comment by ejgallego** [comment]
Thank you @xavierleroy , and congratulations on the 4.07 release; I can confirm that after fixing `C2C.ml` the x64 linux target builds successfully with 4.07.

**Comment by xavierleroy** [comment]
I confirm the C2C.ml fix (commit 693f5b0).
